internal/backend/rest/rest.go

```diff
@@ -246,6 +247,20 @@ func (b *Backend) openReader(ctx context.Context, h restic.Handle, length int, o
 		return nil, errors.Errorf("unexpected HTTP response (%v): %v", resp.StatusCode, resp.Status)
 	}
 
+	// workaround https://github.com/golang/go/issues/46071
+	// see also https://forum.restic.net/t/http2-stream-closed-connection-reset-context-canceled/3743/10
+	if resp.ContentLength == 0 && resp.ProtoMajor == 2 && resp.ProtoMinor == 0 {
+		if clens := resp.Header["Content-Length"]; len(clens) == 1 {
+			if cl, err := strconv.ParseUint(clens[0], 10, 63); err == nil {
+				resp.ContentLength = int64(cl)
+			}
+			if resp.ContentLength != 0 {
```

Comment:
I find this error handling very confusing, mainly because we reuse `resp.ContentLength`.

Reply:
Indeed, whenever `ContentLength` gets changed then it's an error...